### PR TITLE
Don't restart on web resource change

### LINKED ISSUE #6417: Dev mode not working anymore when META-INF/resources is generated
**Describe the bug**
Dev mode is not working correctly when using `./mvnw compile quarkus:dev` on a project where the `target/classes/META-INF/resources` is generated.

image

**Expected behavior**
Should work without any error

**Actual behavior**
It works once every few page refresh

**To Reproduce**

Simulate a Maven step that generate the `target/classes/META-INF/resources`

1. Create a project with Quarkus 1.1.0.Final
2. save the `src/main/resources/META-INF/resources/index.html` somewhere
3. run `rm -rf src/main/resources/META-INF/resources` 
4. run `./mvnw compile quarkus:dev`
5. stop it
6. copy the index.html to `target/classes/META-INF/resources`
7. run `./mvnw compile quarkus:dev`
8. open in browser and refresh multiple times

**Environment (please complete the following information):**
 - Output of `java -version`: reproduced on java 1.8 and java 1.11
 - Quarkus version or git rev: 1.1.0.Final

**Additional context**
This bug was originally met on Code Quarkus using frontend maven plugin to generate the files:
https://github.com/eirslett/frontend-maven-plugin

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Web resources should be modifiable without causing a restart.

Fixes #6417

### CODE REVIEW

**Comment by geoand** [comment]
@ia3andy mind giving this PR as shot?

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java [hunk 1]
```diff
@@ -39,16 +52,21 @@
 @Priority(Priorities.USER + 1)
 public class NotFoundExceptionMapper implements ExceptionMapper<NotFoundException> {
 
+    protected static final String META_INF_RESOURCES_SLASH = "META-INF/resources/";
```
Comment (geoand) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java]: Perhaps avoid the `META-INF/resources` duplication?

**Comment by ia3andy** [comment]
> @ia3andy mind giving this PR as shot?

@geoand sure

**Comment by ia3andy** [comment]
@geoand no more errors 👍

**Comment by geoand** [comment]
Thanks for checking @ia3andy.

PR looks good to me apart for a super minor comment I added

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java (hunk 1 above)
Comment (stuartwdouglas) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java]: Does it really matter? The JVM inlines them so it does not add any overhead.

Comment (geoand) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java]: Of course it doesn't matter :). 
Just as a hygiene thing, nothing more. I'm definitely not going to hold up the PR on this :)

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
The number of this PR is pretty evil, four 6s in a row :P

**Pull request merged**